astropy/table/table.py

```diff
@@ -730,8 +739,8 @@ def _init_from_table(self, data, names, dtype, n_cols, copy):
         """Initialize table from an existing Table object """
 
         table = data  # data is really a Table, rename for clarity
-        self.meta.clear()
-        self.meta.update(deepcopy(table.meta))
+        if table.meta:
```

Comment:
This is one of a number of places where if `table.meta` doesn't evaluate to True in boolean context then just leave the output `meta` unset.  If user accesses it then it will be auto-initialized to an empty OrderedDict, but in the meantime don't waste cycles.  Doing a boolean test is fast (measured in ns) compared to running copy or deepcopy (pure Python code, several usec at least).